Tests/UnitTests/StoreKitExtensions/SKErrorTests.swift

```diff
@@ -0,0 +1,147 @@
+    }
+
+    func testPurchaseNotAllowedError() {
+        let error: SKError = .init(.paymentNotAllowed)
+
+        verifyPurchasesError(error,
+                             expectedCode: .purchaseNotAllowedError,
+                             underlyingError: error)
+    }
+
+    func testPaymentCancelled() {
+        let error: SKError = .init(.paymentCancelled)
+
+        verifyPurchasesError(error,
+                             expectedCode: .purchaseCancelledError,
+                             underlyingError: error)
+    }
+
+    func testPaymentInvalid() {
+        let error: SKError = .init(.paymentInvalid)
+
+        verifyPurchasesError(error,
+                             expectedCode: .purchaseInvalidError,
+                             underlyingError: error)
+    }
+
+    func testProductNotAvailableError() {
+        let error: SKError = .init(.storeProductNotAvailable)
+
+        verifyPurchasesError(error,
+                             expectedCode: .productNotAvailableForPurchaseError,
+                             underlyingError: error)
+    }
+
+    @available(iOS 14.0, tvOS 14.0, macOS 11.0, watchOS 6.2, *)
+    func testUnsupportedPlatformError() throws {
+        try AvailabilityChecks.iOS14APIAvailableOrSkipTest()
+
+        let error: SKError = .init(.unsupportedPlatform)
+
+        verifyPurchasesError(error,
+                             expectedCode: .purchaseNotAllowedError,
+                             underlyingError: error)
+    }
+
+    @available(iOS 14.0, tvOS 14.0, macOS 11.0, watchOS 6.2, *)
+    func testIneligibleForOfferError() throws {
+        try AvailabilityChecks.iOS14APIAvailableOrSkipTest()
+
+        let error: SKError = .init(.ineligibleForOffer)
+
+        verifyPurchasesError(error,
+                             expectedCode: .ineligibleError,
+                             underlyingError: error)
+    }
+
+    @available(iOS 14.0, tvOS 14.0, macOS 11.0, watchOS 6.2, *)
+    func testInvalidOfferError() throws {
+        try AvailabilityChecks.iOS14APIAvailableOrSkipTest()
+
+        let error: SKError = .init(.invalidOfferIdentifier)
+
+        verifyPurchasesError(error,
+                             expectedCode: .invalidPromotionalOfferError,
+                             underlyingError: error)
+    }
+
+    func testCurrentlySubscribedError() throws {
+        let error = try XCTUnwrap(
+            NSError(
+                domain: SKErrorDomain,
+                code: SKError.unknown.rawValue,
+                userInfo: [
+                    NSUnderlyingErrorKey: NSError(
+                        domain: "ASDServerErrorDomain",
+                        code: 3532,
+                        userInfo: [:]
+                    )
+                ]
+            ) as? SKError
+        )
+
+        verifyPurchasesError(error,
+                             expectedCode: .productAlreadyPurchasedError,
+                             underlyingError: error)
+    }
```

Comment:
These are much better to test this way instead of in `PurchasesTests`. Though I left the original tests too.